### PR TITLE
Add shell completions by shtab

### PR DESCRIPTION
**Pull request by Freed-Wu**
```sh
gpustat --print-completion bash | sudo tee /usr/share/bash-completion/completions/gpustat
gpustat --print-completion tcsh | sudo tee /etc/profile.d/gpustat.completion.csh
gpustat --print-completion zsh | sudo tee /usr/share/zsh/site-functions/_gpustat
```

```
% gpustat -i <TAB>
float
% gpustat -i -<TAB>
option
-a                   Display all gpu properties above
...
```

### CODE REVIEW

**Comment by Freed-Wu** [comment]
I fix the conflict.

File: gpustat/cli.py [hunk 1]
```diff
@@ -129,6 +185,8 @@ def nonnegative_int(value):
     parser.add_argument('-v', '--version', action='version',
                         version=('gpustat %s' % __version__))
     args = parser.parse_args(argv[1:])
+    # TypeError: GPUStatCollection.print_formatted() got an unexpected keyword argument 'print_completion'
+    del args.print_completion
```
Comment (wookayin) [inline gpustat/cli.py:189]: Without shtab installed:

```
  File ".../gpustat/gpustat/__main__.py", line 9, in <module> 
    main()
  File ".../gpustat/gpustat/cli.py", line 189, in main 
    del args.print_completion
AttributeError: print_completion
```

File: gpustat/cli.py [hunk 2]
```diff
@@ -69,7 +114,12 @@ def main(*argv):
         pass
     # arguments to gpustat
     import argparse
+    try:
+        import shtab
+    except ImportError:
+        from . import _shtab as shtab
```
Comment (wookayin) [inline gpustat/cli.py:120]: Why do we want to provide `_shtab` fallback when `shtab` is not installed? (please see the errors around L189)  Would you want to make `shtab` as an optional dependency?

**Review by wookayin — CHANGES_REQUESTED** [review]
Thanks, I think the feature works nice but there are some corner cases.

**Comment by wookayin** [comment]
Another corner case: if we execute gpustat as `python -m gpustat`, the completion file will start with `#compdef __main__.py` instead of `#compdef gpustat`. It might be an issue of shtab itself, but do you have any idea about how we can force the program name to be `gpustat`?

File: gpustat/cli.py (hunk 2 above)
Comment (Freed-Wu) [reply gpustat/cli.py:121]: Right :smile:
```python
    extras_require={'test': tests_requires, 'completion': ['shtab']},
```

**Comment by Freed-Wu** [comment]
> Another corner case: if we execute gpustat as `python -m gpustat`, the completion file will start with `#compdef __main__.py` instead of `#compdef gpustat`. It might be an issue of shtab itself, but do you have any idea about how we can force the program name to be `gpustat`?

Fixed. :smile:

```python
    parser = argparse.ArgumentParser(prog='gpustat')
```

**Comment by Freed-Wu** [comment]
Any other problems? It still display `1 review requesting changes`.

**Review by wookayin — APPROVED** [review]
Looks good, thank you. I will merge it soon and write docs for this.

**Comment by wookayin** [comment]
I have rebased your commit due to conflicts. Let me add some minor touches.

**Comment by Freed-Wu** [comment]
Good. When can it be accepted?

**Comment by wookayin** [comment]
@Freed-Wu Thank you for your contribution!

**Comment by wookayin** [comment]
For those who do not have root privileges, you can do like

```
$ pip install gpustat[completion]  # or do pip install shtab

$ gpustat --print-completion zsh > ~/.local/share/zsh/site-functions/_gpustat
$ gpustat --print-completion bash > ~/.bash_completion.d/gpustat
```

and add `$HOME/.local/share/zsh/site-functions/_gpustat` to `$fpath` (for zsh), or [source the completion file](https://serverfault.com/questions/506612/standard-place-for-user-defined-bash-completion-d-scripts) (for bash).
